github/github.go

```diff
@@ -554,7 +554,21 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
 
 		return nil, err
 	}
-	defer resp.Body.Close()
+
+	defer func() {
+		// Ensure the response body is fully read and closed
+		// before we reconnect, so that we reuse the same TCP connection.
+		// Close the previous response's body. But read at least some of
+		// the body so if it's small the underlying TCP connection will be
+		// re-used. No need to check for errors: if it fails, the Transport
+		// won't reuse it anyway.
+		const maxBodySlurpSize = 2 << 10
+		if resp.ContentLength == -1 || resp.ContentLength <= maxBodySlurpSize {
```

Comment:
I'm not sure why I didn't require unit tests on this PR, and that is my fault.

This PR causes a major breakage because it doesn't check if `resp` is `nil` and will therefore panic in those cases.

Please see #1601 for proposed fix.

Reply:
@gmlewis I just saw the issue. I am not sure if that fix is actually required. 



The HTTP client returns the `nil` response whenever any error occurs.
Please check https://golang.org/src/net/http/client.go

In that case, we are already handling errors as you could see here at https://github.com/google/go-github/blob/master/github/github.go#L538

Moreover, The underlying Roundtripper() also checks such possibilities 

By simply checking for an error before doing anything with the response we can conclude what we can and cannot do next.
- if the error is not nil, we have an error. Return now or continue but avoid using the nil response
- if the error is nil, we don’t have an error. We can freely use the response as we originally intended.

Thanks!